app/assets/javascripts/oxalis/controller/viewmodes/arbitrary_controller.js

```diff
@@ -258,42 +241,67 @@ class ArbitraryController {
 
 
   bindToEvents(): void {
-    this.listenTo(this.arbitraryView, "render", this.render);
+    this.listenTo(this.arbitraryView, "render", this.pingBinaries);
+    this.listenTo(this.arbitraryView, "render", this.props.onRender);
 
     for (const name of Object.keys(Model.binary)) {
       const binary = Model.binary[name];
       this.listenTo(binary.cube, "bucketLoaded", this.arbitraryView.draw);
     }
 
-    Store.subscribe(() => {
-      const { sphericalCapRadius, clippingDistanceArbitrary, displayCrosshair } = Store.getState().userConfiguration;
-      this.crosshair.setScale(sphericalCapRadius);
-      this.plane.setMode(this.mode);
-      this.setClippingDistance(clippingDistanceArbitrary);
-      this.crosshair.setVisibility(displayCrosshair);
-    });
+    this.storePropertyUnsubscribers.push(
+      listenToStoreProperty(state => state.userConfiguration, (userConfiguration) => {
+        const { sphericalCapRadius, clippingDistanceArbitrary, displayCrosshair } = userConfiguration;
+        this.crosshair.setScale(sphericalCapRadius);
+        this.setClippingDistance(clippingDistanceArbitrary);
+        this.crosshair.setVisibility(displayCrosshair);
+      }),
+      listenToStoreProperty(state => state.temporaryConfiguration.flightmodeRecording, (isRecording) => {
+        if (isRecording) {
+          this.setWaypoint();
```

Comment:
Minor; but I would prefer to add a comment here, explaining why and when this waypoint is set.
("Add a node directly after switching into recording mode" or something similar...)